### ISSUE TITLE
An increase in loss after shuffling the training set

### ISSUE THREAD

**Issue by faruknane** [microsoft/CNTK#3792]
My training data -> x_train has 32 instances (32 training example).

I train my model with x_train and loss goes to 0 as below.
![image](https://user-images.githubusercontent.com/37745467/73405682-9ad88500-4305-11ea-9e53-7dafa00428c9.png)

After I shuffle my x_train.
![image](https://user-images.githubusercontent.com/37745467/73406325-38808400-4307-11ea-9e58-22e0a79d33c9.png)

Loss goes so high.
![image](https://user-images.githubusercontent.com/37745467/73406356-4e8e4480-4307-11ea-9480-bd6993fac8bf.png)

Why is that? I made sure that shuffling works just fine and I don't understand why this is happening.

My model is like below:
![image](https://user-images.githubusercontent.com/37745467/73405929-4681d500-4306-11ea-9427-4609fcccab6d.png)
